marlowe-playground-client/src/Projects/View.purs

```diff
@@ -0,0 +1,103 @@
+module Projects.View (render) where
+
+import Prelude hiding (div)
+import Data.Array (filter)
+import Data.DateTime.ISO as ISO
+import Data.Either (Either(..))
+import Data.Formatter.DateTime (FormatterCommand(..), format)
+import Data.Lens (to, (^.))
+import Data.List (fromFoldable)
+import Data.Maybe (Maybe(..))
+import Data.Newtype (unwrap)
+import Effect.Aff.Class (class MonadAff)
+import Gist (Gist, gistCreatedAt, gistDescription, gistId, gistUpdatedAt)
+import Halogen (ClassName(..), ComponentHTML)
+import Halogen.Classes (fontSemibold, modalContent, paddingRight, smallPaddingRight, textSm)
+import Halogen.HTML (HTML, a, a_, div, div_, span, text)
+import Halogen.HTML.Events (onClick)
+import Halogen.HTML.Properties (class_, classes)
+import MainFrame.Types (ChildSlots)
+import Marlowe.Gists (fileExists, filenames, playgroundGist)
+import Modal.ViewHelpers (modalHeader)
+import Network.RemoteData (RemoteData(..))
+import Prim.TypeError (class Warn, Text)
+import Projects.Types (Action(..), Lang(..), State, _projects, modalIsLoading)
+import Text.Parsing.Parser (runParser)
+
+render ::
+  forall m.
+  MonadAff m =>
+  State ->
+  ComponentHTML Action ChildSlots m
+render state =
+  if modalIsLoading state then
+    text mempty
+  else
+    div_
+      [ modalHeader "Open Project" (Just Cancel)
+      , div [ classes [ modalContent, ClassName "open-project-modal" ] ]
+          [ body playgroundGists
+          ]
+      ]
+  where
+  playgroundGists = filter playgroundGist <$> state ^. _projects
+
+  body (Success []) = span [ class_ (ClassName "empty-result") ] [ text "No saved projects found" ]
+
+  body (Success gists) = projectList gists
+
+  body (Failure _) = span [ class_ (ClassName "error") ] [ text "Failed to load gists" ]
+
+  body _ = text mempty
+
+projectList ::
+  forall p.
+  Warn (Text "Only 30 projects seem to be loading. Check why and define what should we do about it") =>
+  Array Gist ->
+  HTML p Action
+projectList gists =
+  div [ classes [ ClassName "project-list" ] ]
+    (headers <> projects)
+  where
+  headers :: Array (HTML p Action)
+  headers =
+    [ "Name", "Created", "Last Updated", "Open" ]
+      <#> \name ->
```

Comment:
I personally prefer `for` and I think we should make a decision on this one way or another and be consistent

Reply:
Not sure if you meant this:

```purescript
    for
      [ "Name", "Created", "Last Updated", "Open" ] \name ->
        div [ classes [ textSm, fontSemibold ] ] [ text name ]
```
But that is causing a type error

```
 Could not match type

    Array

  with type

    HTML p0
```

```purescript
for :: forall a b m t. Applicative m => Traversable t => t a -> (a -> m b) -> m (t b)
for :: forall b m. Applicative m => Array String -> (String -> m b) -> m (Array b)
```
And I think `HTML` doesn't have an Applicative instance

Reply:
oh sorry, this is backwards `map` isn't it. Or `flip map`. I guess it's something new for me to get used to, I'm just not a big fan of infix. So ignore this comment